flink/v1.14/build.gradle

```diff
@@ -174,6 +232,16 @@ project(":iceberg-flink:iceberg-flink-runtime-${flinkMajorVersion}") {
     classifier null
   }
 
+  task integrationTest(type: Test) {
+    description = "Test Flink Runtime Jar against Flink ${flinkMajorVersion}"
+    group = "verification"
+    testClassesDirs = sourceSets.integration.output.classesDirs
+    classpath = sourceSets.integration.runtimeClasspath + files(shadowJar.archiveFile.get().asFile.path)
+    inputs.file(shadowJar.archiveFile.get().asFile.path)
+  }
+  integrationTest.dependsOn shadowJar
+  check.dependsOn integrationTest
```

Comment:
the integration test takes < 20 seconds in total, so it should be ok to run this as part of normal CI when the `check` task is running